### PR TITLE
`extra_nest_types` argument for nest functions

### PR DESCRIPTION
**Pull request by sherry30**

### CODE REVIEW

File: ivy/functional/ivy/nest.py
Comment (djl11) [inline ivy/functional/ivy/nest.py:513]: isinstance(nest, (tuple, list) + extra_nest_types)

File: ivy/functional/ivy/nest.py
Comment (djl11) [inline ivy/functional/ivy/nest.py:642]: isinstance(nest, (tuple, list) + extra_nest_types)

File: ivy/functional/ivy/nest.py
Comment (djl11) [inline ivy/functional/ivy/nest.py:879]: isinstance(nest, (tuple, list) + extra_nest_types)

File: ivy/functional/ivy/nest.py
Comment (djl11) [inline ivy/functional/ivy/nest.py:955]: isinstance(nest, (tuple, list) + extra_nest_types)

File: ivy/functional/ivy/nest.py
Comment (djl11) [inline ivy/functional/ivy/nest.py:1053]: isinstance(nest, (tuple, list) + extra_nest_types)

**Review by djl11 — CHANGES_REQUESTED** [review]
small change (not sure I found all cases). Once fixed, then I think it's good to merge! Once merged, could you also please extend the nest unit tests to also test for this new argument? Thanks!

File: ivy/functional/ivy/nest.py
Comment (sherry30) [reply ivy/functional/ivy/nest.py:879]: This one is using `list_check_fn` for checking list type, so isn't this ok?

File: ivy/functional/ivy/nest.py
Comment (sherry30) [reply ivy/functional/ivy/nest.py:1053]: This one is using `check_fn` for checking list type.

File: ivy/functional/ivy/nest.py
Comment (sherry30) [reply ivy/functional/ivy/nest.py:879]: one more thing I realised, `extra_nest_types` can be `Optional[Union[type, Tuple[type]]]`. So it can just be a type and not a tuple of types, in which case doing `(tuple, list) + extra_nest_types` wont work

**Review by djl11 — APPROVED** [review]
agree with your points, looks good to merge!

**Pull request merged**
